prow/crier/reporters/gerrit/reporter.go

```diff
@@ -525,35 +531,90 @@ func (j *Job) serializeWithoutURL() string {
 }
 
 func (j *Job) serialize() string {
-	return fmt.Sprintf(jobReportFormat, j.Icon, j.Name, strings.ToUpper(string(j.State)), j.URL)
-}
+	format := jobReportFormat
 
-func deserializeWithoutURL(s string, j *Job) error {
-	var state string
-	n, err := fmt.Sscanf(s, jobReportFormatWithoutURL, &j.Icon, &j.Name, &state)
-	if err != nil {
-		return err
+	// It may be that the URL is just "URL_NOT_FOUND", so we have to take care
+	// not to link it as such if we're doing Markdown-flavored URLs.
+	if j.URL == urlNotFound {
+		format = jobReportFormatUrlNotFound
 	}
-	j.State = v1.ProwJobState(strings.ToLower(state))
-	const want = 3
-	if n != want {
-		return fmt.Errorf("scan: got %d, want %d", n, want)
-	}
-	return nil
+
+	return fmt.Sprintf(format, j.Icon, j.Name, j.URL, strings.ToUpper(string(j.State)))
 }
 
 func deserialize(s string, j *Job) error {
 	var state string
-	n, err := fmt.Sscanf(s, jobReportFormat, &j.Icon, &j.Name, &state, &j.URL)
-	if err != nil {
-		return err
-	}
-	j.State = v1.ProwJobState(strings.ToLower(state))
-	const want = 4
-	if n != want {
-		return fmt.Errorf("scan: got %d, want %d", n, want)
+	var formats = []struct {
+		regex string
+		// TODO(listx): In Go 1.18, we should replace interface{} with "any".
+		tokens []interface{}
+	}{
+		// Legacy format. This is to cover the case where we're still trying to
+		// parse legacy style comments during the transition to the new style
+		// (just in case).
+		//
+		// TODO(listx): It should be safe to delete this legacy format checker
+		// after we migrate all Prow instances over to the version of crier's
+		// gerrit reporter (this file) that uses the Markdown-flavored links.
+		// There is no hurry to delete this code because having it here is
+		// harmless, other than incurring negligible CPU cycles.
+		{jobReportFormatLegacyRegex,
+			[]interface{}{&j.Icon, &j.Name, &state, &j.URL}},
```

Comment:
these are all string pointers right? can we use `[]*string` instead of interface, so that there is no need to do the `(*string)` cast dance below?

Reply:
Done